scene/main/node.cpp

```diff
@@ -994,8 +994,28 @@ void Node::_validate_child_name(Node *p_child, bool p_force_human_readable) {
 
 		if (!unique) {
 			ERR_FAIL_COND(!node_hrcr_count.ref());
-			String name = "@" + String(p_child->get_name()) + "@" + itos(node_hrcr_count.get());
-			p_child->data.name = name;
+			// Optimized version of unique name generation.
+			uint32_t c = node_hrcr_count.get();
+			String cn = p_child->get_class_name().operator String();
+			const char32_t *cn_ptr = cn.ptr();
+			uint32_t cn_length = cn.length();
+			uint32_t c_chars = String::num_characters(c);
+			uint32_t len = 2 + cn_length + c_chars;
+			char32_t *str = (char32_t *)alloca(sizeof(char32_t) * (len + 1));
```

Comment:
This buffer isn't deleted (the string constructor doesn't take ownership of it)

Also would it make sense to use a thread static shared buffer of like size 64 or so and only allocate this temporary buffer if that is too small?

Alternatively make this buffer a `CowData<char32_t>` and add a `String` constructor that takes ownership of it.

Reply:
[alloca](https://man7.org/linux/man-pages/man3/alloca.3.html)() allocates on the stack. It's the most efficient way of allocation for this kind of usage and it does not need to be freed.

Reply:
Or you can just simplest solution that is already implemented; nvm then